src/mame/atari/a2600.cpp

```diff
@@ -481,6 +481,11 @@ void a2600_state::machine_start()
 		break;
 	case A26_SS:
 		m_maincpu->space(AS_PROGRAM).install_read_handler(0x1000, 0x1fff, read8sm_delegate(*m_cart, FUNC(vcs_cart_slot_device::read_rom)));
+		// The SuperCharger monitors the entire address bus for changes
+		m_maincpu->space(AS_PROGRAM).install_readwrite_tap(0x0000, 0x1fff, "tap",
+			[this](offs_t offset, u8 &, u8) { if(!machine().side_effects_disabled()) m_cart->tap(offset); },
+			[this](offs_t offset, u8 &, u8) { if(!machine().side_effects_disabled()) m_cart->tap(offset); }
+		);
```

Comment:
This is a “lipstick on a pig” approach, and too inflexible.  The A2600 cartridges are a leaky abstraction – the host should not know where to install cartridge handlers like this.  I’d rather see it refactored so the cartridges are just supplied with the space to install to, allowing them to install their own taps if they need to rather than continue on a dead-end route that needs to be replaced anyway.

Reply:
And as a side effect that would also remove the trampolines through the cart_slot_device, makes sense.